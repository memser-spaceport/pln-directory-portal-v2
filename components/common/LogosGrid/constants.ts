export const LOGOS = [
  '/icons/demoday/landing/logos/Protocol Labs.svg',
  '/icons/demoday/landing/logos/A16zcrypto.svg',
  '/icons/demoday/landing/logos/Coinbase.svg',
  '/icons/demoday/landing/logos/HashKey.svg',
  '/icons/demoday/landing/logos/BlueYard.svg',

  '/icons/demoday/landing/logos/Multicoin.svg',
  '/icons/demoday/landing/logos/PolychainCapital.svg',
  '/icons/demoday/landing/logos/Y.svg',
  '/icons/demoday/landing/logos/PanteraCapital.svg',
  '/icons/demoday/landing/logos/GreyCroft.svg',

  '/icons/demoday/landing/logos/KleinerPerkins.svg',
  '/icons/demoday/landing/logos/Redpoint.svg',
  '/icons/demoday/landing/logos/UnionVentures.svg',
  '/icons/demoday/landing/logos/Venrock.svg',
  '/icons/demoday/landing/logos/Galaxy.svg',

  '/icons/demoday/landing/logos/InsightPartners.svg',
  '/icons/demoday/landing/logos/JuanBenet.svg',
  '/icons/demoday/landing/logos/BalajiSrinivasan.svg',
  '/icons/demoday/landing/logos/VitalikButerin.svg',
  '/icons/demoday/landing/logos/AnatolyYakovenko.svg',
  '/icons/demoday/landing/logos/DigitalCurrency.svg',

  '/icons/demoday/landing/logos/FenbushiCapital.svg',
  '/icons/demoday/landing/logos/Hashed.svg',
  '/icons/demoday/landing/logos/TribeCapital.svg',
  '/icons/demoday/landing/logos/Jump.svg',
  '/icons/demoday/landing/logos/Sosv.svg',

  '/icons/demoday/landing/logos/CoinFund.svg',
  '/icons/demoday/landing/logos/Amber.svg',
  '/icons/demoday/landing/logos/AngelList.svg',
  '/icons/demoday/landing/logos/AnimocaBrands.svg',
  '/icons/demoday/landing/logos/DelphiVentures.svg',

  '/icons/demoday/landing/logos/GGVCapital.svg',
  '/icons/demoday/landing/logos/IOSGVentures.svg',
  '/icons/demoday/landing/logos/Republic.svg',
  '/icons/demoday/landing/logos/SolanaVentures.svg',
  '/icons/demoday/landing/logos/Spartan.svg',

  '/icons/demoday/landing/logos/SVAngel.svg',
  '/icons/demoday/landing/logos/ElectricCapital.svg',
  '/icons/demoday/landing/logos/OrangeDAO.svg',
  '/icons/demoday/landing/logos/Alliance.svg',
  '/icons/demoday/landing/logos/BigBrainHolding.svg',

  '/icons/demoday/landing/logos/Collabcurrency.svg',
  '/icons/demoday/landing/logos/Mesh.svg',
  '/icons/demoday/landing/logos/Placeholder.svg',
  '/icons/demoday/landing/logos/ShimaCapital.svg',
  '/icons/demoday/landing/logos/NorthZone.svg',

  '/icons/demoday/landing/logos/FJLabs.svg',
  '/icons/demoday/landing/logos/Dialectic.svg',
  '/icons/demoday/landing/logos/Arca.svg',
  '/icons/demoday/landing/logos/Borderless.svg',
  '/icons/demoday/landing/logos/MechanismCapital.svg',

  '/icons/demoday/landing/logos/HiveMind.svg',
  '/icons/demoday/landing/logos/FabricVentures.svg',
  '/icons/demoday/landing/logos/Framework.svg',
  '/icons/demoday/landing/logos/JustinKan.svg',
  '/icons/demoday/landing/logos/ChrisLarsen.svg',

  '/icons/demoday/landing/logos/GSR.svg',
  '/icons/demoday/landing/logos/Techstars.svg',
  '/icons/demoday/landing/logos/LongHashVentures.svg',
  '/icons/demoday/landing/logos/Eniac.svg',
  '/icons/demoday/landing/logos/LDCapital.svg',

  '/icons/demoday/landing/logos/NIV.svg',
  '/icons/demoday/landing/logos/LightspeedAction.svg',
  '/icons/demoday/landing/logos/CmtDigital.svg',
  '/icons/demoday/landing/logos/Archetype.svg',
  '/icons/demoday/landing/logos/KrakenVentures.svg',

  '/icons/demoday/landing/logos/OutlinerVentures.svg',
];
