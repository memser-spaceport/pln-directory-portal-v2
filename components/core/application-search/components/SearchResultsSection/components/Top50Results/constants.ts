export const MAX_ITEMS_TO_SHOW = 5;
