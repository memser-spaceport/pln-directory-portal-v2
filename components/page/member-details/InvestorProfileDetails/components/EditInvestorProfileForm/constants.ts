export const investorTypeOptions = [
  { label: 'I angel invest', value: 'ANGEL' },
  { label: 'I invest through fund(s)', value: 'FUND' },
  { label: 'I angel invest + invest through fund(s)', value: 'ANGEL_AND_FUND' },
];

export const investInVcFundsOptions = [
  { label: 'Early stage', value: 'early-stage' },
  { label: 'Late stage', value: 'late-stage' },
  { label: 'Fund-of-funds', value: 'fund-of-funds' },
];

export const SEC_RULES_URL = 'https://www.investor.gov/introduction-investing/general-resources/news-alerts/alerts-bulletins/investor-bulletins/updated-3';
export const EMAIL_SETTINGS_URL = '/settings/email';
export const TEAMS_ADD_URL = '/teams/add';
