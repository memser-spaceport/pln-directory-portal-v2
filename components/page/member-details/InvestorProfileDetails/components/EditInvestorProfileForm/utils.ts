/**
 * Formats a number to currency format
 */
export const formatNumberToCurrency = (value?: string | number): string => {
  if (!value) return '';

  // Convert to number if it's a string
  const numericValue = typeof value === 'string' ? parseFloat(value) : value;
  
  // Return empty string if not a valid number
  if (isNaN(numericValue)) return '';

  // Format as currency
  return new Intl.NumberFormat('en-US', {
    style: 'currency',
    currency: 'USD',
    minimumFractionDigits: 0,
    maximumFractionDigits: 2,
  }).format(numericValue);
};
