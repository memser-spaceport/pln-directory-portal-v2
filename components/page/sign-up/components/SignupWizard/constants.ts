export const TERMS_OF_SERVICE_AND_PRIVACY_URL =
  'https://docs.google.com/document/d/1J1rYRvnkqzXOnbBoQ3PzjIrSZOTufAL4CJPoOBDq-fE/edit';
