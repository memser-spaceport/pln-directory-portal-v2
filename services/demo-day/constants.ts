export enum DemoDayQueryKeys {
  GET_DEMO_DAY_STATE = 'GET_DEMO_DAY_STATE',
  GET_FUNDRAISING_PROFILE = 'GET_FUNDRAISING_PROFILE',
  GET_TEAMS_LIST = 'GET_TEAMS_LIST',
  GET_ALL_FUNDRAISING_PROFILES = 'GET_ALL_FUNDRAISING_PROFILES',
  GET_ENGAGEMENT = 'GET_ENGAGEMENT',
}
