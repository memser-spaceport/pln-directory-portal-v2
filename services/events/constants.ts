export enum EventsQueryKeys {
  GET_UPCOMING_EVENTS = 'GET_UPCOMING_EVENTS',
}
