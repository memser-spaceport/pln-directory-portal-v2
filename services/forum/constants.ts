export enum ForumQueryKeys {
  GET_CATEGORIES = 'GET_CATEGORIES',
  GET_TOPICS = 'GET_TOPICS',
  GET_TOPIC = 'GET_TOPIC',
  GET_VOTE_STATUS = 'GET_VOTE_STATUS',
}
