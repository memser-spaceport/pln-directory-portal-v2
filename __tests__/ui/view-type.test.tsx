import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import ViewType from '@/components/ui/view-type';
import { VIEW_TYPE_OPTIONS } from '@/utils/constants';

// Mock next/image to render a simple img
jest.mock('next/image', () => (props: any) => <img {...props} />);
// Mock Tooltip to render its trigger directly
jest.mock('@/components/core/tooltip/tooltip', () => ({ Tooltip: ({ trigger }: any) => trigger }));

describe('ViewType', () => {
  const callback = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders grid and list buttons', () => {
    const { getAllByRole } = render(
      <ViewType view={VIEW_TYPE_OPTIONS.GRID} callback={callback} />
    );
    const buttons = getAllByRole('button');
    expect(buttons.length).toBe(2);
  });

  it('highlights grid as selected when view is GRID', () => {
    const { container } = render(
      <ViewType view={VIEW_TYPE_OPTIONS.GRID} callback={callback} />
    );
    const gridBtn = container.querySelector('.view-option-container__grid');
    expect(gridBtn).toHaveClass('selected');
    const listBtn = container.querySelector('.view-option-container__list');
    expect(listBtn).toHaveClass('not-selected');
  });

  it('highlights list as selected when view is LIST', () => {
    const { container } = render(
      <ViewType view={VIEW_TYPE_OPTIONS.LIST} callback={callback} />
    );
    const gridBtn = container.querySelector('.view-option-container__grid');
    expect(gridBtn).toHaveClass('not-selected');
    const listBtn = container.querySelector('.view-option-container__list');
    expect(listBtn).toHaveClass('selected');
  });

  it('renders correct grid icon for selected/deselected', () => {
    const { container, rerender } = render(
      <ViewType view={VIEW_TYPE_OPTIONS.GRID} callback={callback} />
    );
    let gridImg = container.querySelector('.view-option-container__grid img');
    expect(gridImg).toHaveAttribute('src', '/icons/grid-selected.svg');
    rerender(<ViewType view={VIEW_TYPE_OPTIONS.LIST} callback={callback} />);
    gridImg = container.querySelector('.view-option-container__grid img');
    expect(gridImg).toHaveAttribute('src', '/icons/grid-deselect.svg');
  });

  it('renders correct list icon for selected/deselected', () => {
    const { container, rerender } = render(
      <ViewType view={VIEW_TYPE_OPTIONS.LIST} callback={callback} />
    );
    let listImg = container.querySelector('.view-option-container__list img');
    expect(listImg).toHaveAttribute('src', '/icons/list-selected.svg');
    rerender(<ViewType view={VIEW_TYPE_OPTIONS.GRID} callback={callback} />);
    listImg = container.querySelector('.view-option-container__list img');
    expect(listImg).toHaveAttribute('src', '/icons/list-deselect.svg');
  });

  it('calls callback with GRID when grid button is clicked', () => {
    const { container } = render(
      <ViewType view={VIEW_TYPE_OPTIONS.LIST} callback={callback} />
    );
    const gridBtn = container.querySelector('.view-option-container__grid');
    fireEvent.click(gridBtn!);
    expect(callback).toHaveBeenCalledWith(VIEW_TYPE_OPTIONS.GRID);
  });

  it('calls callback with LIST when list button is clicked', () => {
    const { container } = render(
      <ViewType view={VIEW_TYPE_OPTIONS.GRID} callback={callback} />
    );
    const listBtn = container.querySelector('.view-option-container__list');
    fireEvent.click(listBtn!);
    expect(callback).toHaveBeenCalledWith(VIEW_TYPE_OPTIONS.LIST);
  });
}); 