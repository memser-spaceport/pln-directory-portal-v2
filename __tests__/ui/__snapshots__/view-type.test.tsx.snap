// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewType matches snapshot 1`] = `
<div>
  <div
    class="jsx-52b22ea04ef6becf view-option-container"
  >
    <button
      class="jsx-52b22ea04ef6becf view-option-container__grid selected"
    >
      <img
        alt="grid"
        height="24"
        loading="lazy"
        src="/icons/grid-selected.svg"
        width="24"
      />
    </button>
    <button
      class="jsx-52b22ea04ef6becf view-option-container__list not-selected"
    >
      <img
        alt="list"
        height="24"
        loading="lazy"
        src="/icons/list-deselect.svg"
        width="24"
      />
    </button>
  </div>
</div>
`;
